lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -63,6 +67,38 @@ def cache_name(self):
     def cache_path(self):
         return os.path.join(self.pkg.stage.source_path, self.cache_name)
 
+    # Implement a version of the define_from_variant for Cached packages
+    def define_cmake_cache_from_variant(self, cmake_var, variant=None, comment=""):
+        """Return a Cached CMake field from the given variant's value.
+        See define_from_variant in lib/spack/spack/build_systems/cmake.py package
+        """
+
+        if variant is None:
+            variant = cmake_var.lower()
+
+        if variant not in self.pkg.variants:
+            raise KeyError('"{0}" is not a variant of "{1}"'.format(variant, self.pkg.name))
+
+        if variant not in self.pkg.spec.variants:
+            return ""
+
+        value = self.pkg.spec.variants[variant].value
+        if isinstance(value, (tuple, list)):
+            # Sort multi-valued variants for reproducibility
+            value = sorted(value)
```

Comment:
We should already be doing this for all variants (regardless of build system) - are you seeing counterexamples to this?

Reply:
Removed.